sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -849,6 +886,11 @@ def fit(self, X, y, sample_weight=None):
                     "[{}/{}] ".format(iteration + 1, self.max_iter), end="", flush=True
                 )
 
+            # Do out of bag if required
+            if do_oob:
+                self._bagging_subsample_rng.shuffle(sample_mask)
+                sample_weight_train = sample_weight_train_original * sample_mask
```

Comment:
Using null `sample_weight` to mathematically simulate subsampling it's simple to implement and does not need allocating any extra memory but it does not produce the expected 2x computational speed-up of when using a typical `subsample=0.5`.

Have considered row-wise fancy indexing of the training data instead?